### PR TITLE
V0.6.5 forum model2

### PR DESCRIPTION
**Pull request by csoler**

### CODE REVIEW

**Comment by felisucoibi** [comment]
+1000

File: libretroshare/src/retroshare/rsgxsforums.h [hunk 1]
```diff
@@ -150,6 +150,19 @@ class RsGxsForums: public RsGxsIfaceHelper
...
+	 * @brief Get specific list of messages from a single forums. Blocking API
+	 * @jsonapi{development}
+	 * @param[in] forumId id of the forum of which the content is requested
+	 * @param[in] msgs_to_request list of message ids to request
+	 * @param[out] msgs storage for the forum messages
+	 * @return false if something failed, true otherwhise
+	 */
+	virtual bool getForumsContent(
+            const RsGxsGroupId& forumId,
+            std::set<RsGxsMessageId>& msgs_to_request,
```
Comment (G10h4ck) [inline libretroshare/src/retroshare/rsgxsforums.h:178]: please use camel case for param name to be consistent with the other JSON API calls, also in JSON API we can't expose different methods with same name, so this need to have a different name too

File: retroshare-gui/src/gui/gxsforums/GxsForumThreadWidget.cpp [hunk 2]
```diff
@@ -2542,26 +2543,129 @@ bool GxsForumThreadWidget::filterItem(QTreeWidgetItem *item, const QString &text
...
+        RsGxsForumGroup group = groups[0];
+
+        // 3 - update the model in the UI thread.
+
+        RsQThreadUtils::postToObject( [group,this]()
+		{
+			/* Here it goes any code you want to be executed on the Qt Gui
+			 * thread, for example to update the data model with new information
+			 * after a blocking call to RetroShare API complete, note that
+			 * Qt::QueuedConnection is important!
```
Comment (G10h4ck) [inline retroshare-gui/src/gui/gxsforums/GxsForumThreadWidget.cpp:1931]: The QueuedConnection part of the comment is not relevant anymore, so it should be removed ;)

**Review by G10h4ck — CHANGES_REQUESTED** [review]

File: retroshare-gui/src/gui/gxsforums/GxsForumThreadWidget.cpp [hunk 3]
```diff
@@ -2424,40 +2431,60 @@ void GxsForumThreadWidget::flagperson()
...
+
+        msgs_to_request.insert(mThreadId);
+
+		if(!rsGxsForums->getForumsContent(groupId(),msgs_to_request,msgs))
+		{
+			std::cerr << __PRETTY_FUNCTION__ << " failed to retrieve forum group info for forum " << groupId() << std::endl;
+			return;
+        }
+
+        if(msgs.size() != 1)
```
Comment (G10h4ck) [inline retroshare-gui/src/gui/gxsforums/GxsForumThreadWidget.cpp:1662]: This seems an error condition, so a message should be printed ;)

**Comment by csoler** [comment]
3 things left: sorting (and post pinning), post versions, proper fonts for read/unread to match old system. Apart from that the whole thing seems to work pretty well.

**Comment by csoler** [comment]
Still stuck with a bug: sorting does not work. Seems to be a GUI problem. When clicking on the header's columns, the Qt sorting method is not called.

**Comment by felisucoibi** [comment]
testing but seems forums are not loading.

**Comment by felisucoibi** [comment]
Something is wrong, i know this is not a good way to send you a bug report, but just tell you that rs is crashing, i will try to find where or why but It crashes when entering in forum new posts.

**Comment by csoler** [comment]
I removed the [WIP] statement because it seems final to me. Please test.

**Comment by felisucoibi** [comment]
when you choose a forum and no post is selected, before you click in next message and you went to the first last message posted or kind of, now it does nothing, you have to choose a thread first... this feature was usefull to reach fast to new posts. @Cyril

**Comment by felisucoibi** [comment]
and also the icons are good but maybe they need to be a little more bigger, in a 2K screen are very small, i can't imagine in a 3k monitor.@Cyril

**Comment by felisucoibi** [comment]
and the most important forums still don't load, forum posts... i can see the thread but when i click in a new post it does not load. when i click in a forum post this log appears...
(WW) Checking group signature with old api version 2 : tag 0 result: 1
Clicked on message ID bb6af3843588be5f0480d390c7d51986b0071d9f, index=(0,0,0x555db8845eb0)
  early return because mUpdating=true

I think the forum is LOADING and never ends, and that is the reason why.... is says loading on top of forum.

**Comment by felisucoibi** [comment]
i think more or less is ready for open betateting in master, so we can have a beta widows release and my windows users can test it.

**Pull request merged**